### PR TITLE
Rebase of #19222: JME python configuration cleanup for 92x

### PR DESCRIPTION
**Pull request by rappoccio**
Had a git merge problem that I could not fix in #19222, so re-adding a new PR. Here is the original message:

The JME python configurations had many vestiges left of previous migrations, as well as considerable "top-level" customization that is no longer relevant. I have consolidated the RecoPFJets_cff customizations into separate cfi python files in RecoJets/JetProducers.

Removed iterative cone, SIScone, and AK5 PF and Calo jets in all configurations.
Consolidated the various grooming schemes into a single file per jet radius (so, AK4 puppi is now in ak4PFJets_cfi, for instance).
Left AK4, AK8, KT4, KT6, and only AK5/7 Castor collections since those are the only ones used anywhere in the repository.
Added an example AK8 GenJet configuration with grooming applied, since this is a common question. It is not presently run anywhere, though.
EDIT: No changes should occur to any output, this is a reorganization.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by rappoccio** [comment]
Looking at this, there seem to be some O(10%) differences in soft drop. I will follow up.

File: RecoJets/JetProducers/python/ak4PFJets_cfi.py [hunk 1]
```diff
@@ -11,3 +11,25 @@
     rParam       = cms.double(0.4)
```
Comment (ahinzmann) [inline RecoJets/JetProducers/python/ak4PFJets_cfi.py]: How about turning on "doAreaFastjet=True" for ak4PFJets (and thus all other derived ak jet algorithms) during this cleanup. This is currently done only in the PAT/MiniAOD configuration and lead to many errors in the past.

Comment (rappoccio) [inline RecoJets/JetProducers/python/ak4PFJets_cfi.py]: OK, will do.

**Comment by rappoccio** [comment]
OK I figured out what the remaining problem it. It's actually technically a bug in the previous implementation. The AK8 CHS soft dropped jets were previously only requiring pt > 100 GeV, whereas it was intended to also have a pre-clustering with AK8 that has an |eta| < 2.4 cut. 

@ahinzmann @jdolen do you two agree with the addition of the |eta| < 2.4 cut and the AK8 preclustering? 

Our new implementation will slightly speed things up, and fix this "unintended feature". @slava77 is that acceptable?

**Comment by slava77** [comment]
On 6/23/17 8:15 AM, rappoccio wrote:
> OK I figured out what the remaining problem it. It's actually
> technically a bug in the previous implementation. The AK8 CHS soft
> dropped jets were previously only requiring pt > 100 GeV, whereas it was
> intended to also have a pre-clustering with AK8 that has an |eta| < 2.4 cut.
> 
> @ahinzmann <https://github.com/ahinzmann> @jdolen
> <https://github.com/jdolen> do you two agree with the addition of the
> |eta| < 2.4 cut and the AK8 preclustering?
> 
> Our new implementation will slightly speed things up, and fix this
> "unintended feature". @slava77 <https://github.com/slava77> is that
> acceptable?

with so many changes in code lines, I'd rather decouple refactoring from
physics change.

**Comment by rappoccio** [comment]
@slava77 understood. I will retain the old functionality.

**Comment by rappoccio** [comment]
I tried to follow the instructions [here](http://cms-sw.github.io/tutorial-resolve-conflicts.html) to do my edits but apparently it does not like that when I push? I did a pull and then a push, and apparently I'm now bringing in a lot of extra junk in the rebase. What should I do to fix that?

**Comment by rappoccio** [comment]
Never mind, I got it using "-f" to force the push. Woohoo, so proud. ;)

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
please test
(Baseline for previous tests was CMSSW_9_2_X_2017-06-24-1100, which still included the buggy PR 19194, now reverted in CMSSW_9_2_X_2017-06-25-0000)

**Comment by rappoccio** [comment]
Are these failures really due to this PR? I don't see any changes in Physics/B2G or JetMET, where they would show up. I see changes in the TimerService and AlcaBeamMonitor. 

These show failures but I cannot see them?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_2_X_2017-06-25-0000+19406/20872/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/BDHadronTracks_JetContent.html

**Comment by slava77** [comment]
On 6/25/17 11:51 AM, rappoccio wrote:
> Are these failures really due to this PR? I don't see any changes in
> Physics/B2G or JetMET, where they would show up. I see changes in the
> TimerService and AlcaBeamMonitor.
> 
> These show failures but I cannot see them?
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_2_X_2017-06-25-0000+19406/20872/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/BDHadronTracks_JetContent.html
> 

There were no failures this time
However
"""
Reco comparison results: 25 differences found in the comparisons
"""
these appear to be in 2023 workflows

e.g.
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_2_X_2017-06-25-0000+19406/20872/validateJR/all_OldVSNew_TTbar14TeV2023D17wf27434p0/
all in ak8PFJetsCHSPruned

**Comment by rappoccio** [comment]
OK I will take a look. 

cheers,
Sal

**Comment by rappoccio** [comment]
OK this should be fixed. The jet pruning algorithm's definition of "subjet" is not IRC-safe, technically speaking (this affects only the breakdown of the two subjets, not the jet itself). This fixes the errors in, say, WF 27434.0.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
the errors appear to be unrelated to this PR

```
#5  std::string::_M_data (this=0x14e) at /mnt/build/davidlt/gcc530_final/b/BUILD/slc6_amd64_gcc530/external/gcc/5.3.0/gcc-tags_gcc_5_3_0_release-231255/obj/x86_64-pc-linux-gnu/libstdc++-v3/include/bits/basic_string.h:2691
#6  std::string::_M_rep (this=0x14e) at /mnt/build/davidlt/gcc530_final/b/BUILD/slc6_amd64_gcc530/external/gcc/5.3.0/gcc-tags_gcc_5_3_0_release-231255/obj/x86_64-pc-linux-gnu/libstdc++-v3/include/bits/basic_string.h:2699
#7  std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string (this=0x7fe98f765030, __str=...) at /mnt/build/davidlt/gcc530_final/b/BUILD/slc6_amd64_gcc530/external/gcc/5.3.0/gcc-tags_gcc_5_3_0_release-231255/obj/x86_64-pc-linux-gnu/libstdc++-v3/include/bits/basic_string.tcc:617
#8  0x00007fe991872fd5 in XrdCl::URL::GetHostId (this=<optimized out>) at /build/cmsbld/auto-builds/CMSSW_9_1_0_pre3-slc6_amd64_gcc530/build/CMSSW_9_1_0_pre3-build/BUILD/slc6_amd64_gcc530/external/xrootd/4.6.1.RC2/xrootd-4.6.1.RC2/src/./XrdCl/XrdClURL.hh:72

```

@Dr15Jones is this a known xrootd failure?

**Comment by Dr15Jones** [comment]
I can't say that the failure looks familiar. @bbockelm ?

**Comment by perrotta** [comment]
please test
(let see what happens with the new 93X IB)

**Comment by perrotta** [comment]
+1
Reorganization and clean up of jet configurations
After the latest fixes, there is no effect any more on the tested workflows

**Comment by davidlange6** [comment]
merge

**Pull request merged**